pkg/utils/jolokia_client/jolokia_client.go

```diff
@@ -80,7 +83,16 @@ func GetBrokers(resource types.NamespacedName, ssInfos []ss.StatefulSetInfo, cli
 					jolokiaUser, jolokiaPassword, jolokiaProtocol := resolveJolokiaRequestParams(resource.Namespace, client, client.Scheme(), jolokiaSecretName, &containers, podNamespacedName, statefulset, info.Labels)
 
 					reqLogger.Info("New Jolokia with ", "User: ", jolokiaUser, "Protocol: ", jolokiaProtocol, "broker ip", pod.Status.PodIP)
-					artemis := mgmt.GetArtemis(pod.Status.PodIP, "8161", "amq-broker", jolokiaUser, jolokiaPassword, jolokiaProtocol)
+
+					hostToUse, err := resolvePodHostName(pod.Status.PodIP, pod.Name)
+					if err != nil {
+						reqLogger.V(1).Info("failed dns lookup, use IP address", "podIp", pod.Status.PodIP, "error", err)
+						hostToUse = pod.Status.PodIP
+					}
```

Comment:
According to the [Kubernetes DNS documentation](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-hostname-and-subdomain-fields) we should be able to build the hostToUse from the pod hostname and subdomain, i.e.
```suggestion
					hostToUse := pod.Spec.Hostname + pod.Spec.Subdomain + pod.Namespace
```

Reply:
Yes we can have the hostName and subdomain, but I couldn't find where we can get the domain name (default is .cluster.local). The hostname we currently using takes the form <pod-hostname>.<subdomain>.<namespace>.<domain-name>.  Using DNS lookup I think we can get them all.

Reply:
I think that we don't need the domain-name inside the cluster

Reply:
Yes but having it there does no harm. If we don't need the domain we need to change the launch.sh script and drain.sh script because they are using `'hostname -f` to get the full name.

Reply:
another point is that we should trust DNS look up, instead of manually assemble the host name. It's safe for us.

Reply:
just thought that in the cert the SAN names requires the domain name.

Reply:
> Yes but having it there does no harm. If we don't need the domain we need to change the launch.sh script and drain.sh script because they are using `'hostname -f` to get the full name.

I think that having the domain in the hostname could increase the difficulty on creating a certificate, so if not required I would remove it from the hostname but I don't have a strong opinion on that.

> another point is that we should trust DNS look up, instead of manually assemble the host name. It's safe for us.

How could we be sure that `hostname -f` and `resolvePodHostName` will return the same result?
If we build it with the same Kubernetes rules we should be sure that those matches and could provide instructions to users how to create their certificates.

> just thought that in the cert the SAN names requires the domain name.

If we use `<pod-hostname>.<subdomain>` as hostname then `*.<subdomain>` should work
and if we use `<pod-hostname>.<subdomain>.<namespace>.<cluster-domain>` as hostname then `*.<subdomain>.<namespace>.<cluster-domain>` should work

Reply:
Re: How could we be sure that hostname -f and resolvePodHostName will return the same result?
If we build it with the same Kubernetes rules we should be sure that those matches and could provide instructions to users how to create their certificates.

We know hostname -f will return one entry and resolvePosHostName will pick up from the DNS lookup list the one matches it. 

I think we should use the full qualified host name, even if the short one works.